CONTRIBUTING.md

```diff
@@ -25,42 +25,41 @@ install that exact version of rustc as a toolchain:
 ```
 ./rustup-toolchain
 ```
+This will set up a rustup toolchain called `miri` and set it as an override for
+the current directory.
 
 [`rustup-toolchain-install-master`]: https://github.com/kennytm/rustup-toolchain-install-master
 
-### Fixing Miri when rustc changes
+## Building and testing Miri
 
-Miri is heavily tied to rustc internals, so it is very common that rustc changes
-break Miri.  Fixing those is a good way to get starting working on Miri.
-Usually, Miri will require changes similar to the other consumers of the changed
-rustc API, so reading the rustc PR diff is a good way to get an idea for what is
-needed.
+Invoking Miri requires getting a bunch of flags right and setting up a custom
+sysroot with xargo. The `miri` script takes care of that for you. With the
+build environment prepared, compiling Miri is just one command away:
 
-To update the `rustc-version` file and install the latest rustc, you can run:
 ```
-./rustup-toolchain HEAD
+./miri build
 ```
 
-Now try `./miri test`, and submit a PR once that works again.
+Run `./miri` without arguments to see the other commands our build tool
+supports.
 
-## Testing the Miri driver
-[testing-miri]: #testing-the-miri-driver
+### Testing the Miri driver
 
-The Miri driver in the `miri` binary is the "heart" of Miri: it is basically a
-version of `rustc` that, instead of compiling your code, runs it.  It accepts
-all the same flags as `rustc` (though the ones only affecting code generation
-and linking obviously will have no effect) [and more][miri-flags].
+The Miri driver in the `src/bin/miri.rs` binary is the "heart" of Miri: it is
```

Comment:
"the `src/bin/miri.rs` binary" sounds strange to me - is it the binary made from ...?

Reply:
I adjusted the wording, better like this?

> The Miri driver compiled from `src/bin/miri.rs` is the "heart" of Miri: